Sources/Kitura/staticFileServer/FileServer.swift

```diff
@@ -187,9 +187,15 @@ extension StaticFileServer {
                         // Send only headers
                         _ = response.send(status: .OK)
                     } else {
-                        // Send the entire file
-                        try response.send(fileName: filePath)
-                        response.statusCode = .OK
+                        if let etag = request.headers["If-None-Match"],
+                          etag == CacheRelatedHeadersSetter.calculateETag(from: fileAttributes) {
+                            response.statusCode = .notModified
+                            response.send("")
```

Comment:
I don't think this empty `send("")` is necessary - it'll do some work under the covers, but result in no data being appended to the response buffer.  Setting the `statusCode` is sufficient.

One of the outcomes of invoking `send()` is that it sets `response.state.invokedSend` to indicate that a response has/is being sent.  However, we only send a default response if `invokedSend` is false _and_ `statusCode` is `.unknown`.